commit b5d340799db947dc096f4d56c3ea6edb1f453782
Author: hasufell <hasufell@gentoo.org>
Date:   Thu Jul 18 13:58:22 2013 +0200

    BUILD: don't overwrite env ldflags and make as-needed optional

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d52a46c..0d375ec 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,6 +7,7 @@ project(whiskermenu)
 
 # options
 option(ENABLE_WERROR "Enable -Werror flag for development" OFF)
+option(ENABLE_AS_NEEDED "Enable -Wl,--as-needed for the linker" ON)
 
 if(NOT DEFINED LIB_INSTALL_DIR)
     if (DEFINED CMAKE_INSTALL_LIBDIR)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 26972af..dc563c0 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -4,7 +4,9 @@ if(ENABLE_WERROR)
 endif()
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${WHISKERMENU_FLAGS}")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fvisibility=hidden -std=c++0x ${WHISKERMENU_FLAGS}")
-set(CMAKE_SHARED_LINKER_FLAGS "-Wl,--as-needed")
+if(ENABLE_AS_NEEDED)
+	set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--as-needed")
+endif()
 
 find_package(GTK2 REQUIRED)
 find_package(PkgConfig REQUIRED)
